google/devtools/clouddebugger/v2/clouddebugger_gapic.yaml

```diff
@@ -0,0 +1,167 @@
+type: com.google.api.codegen.ConfigProto
+language_settings:
+  java:
+    package_name: com.google.cloud.devtools.clouddebugger.spi.v2
+  python:
+    package_name: google.cloud.devtools.clouddebugger.v2
+  go:
+    package_name: cloud.google.com/go/devtools/clouddebugger/apiv2
+  csharp:
+    package_name: Google.Devtools.Clouddebugger.V2
+  ruby:
+    package_name: Google::Cloud::Devtools::Clouddebugger::V2
+  php:
+    package_name: Google\Cloud\Devtools\Clouddebugger\V2
+  nodejs:
+    package_name: '@google-cloud/clouddebugger'
+interfaces:
+- name: google.devtools.clouddebugger.v2.Debugger2
+  collections: []
+  retry_codes_def:
+  - name: idempotent
+    retry_codes:
+    - UNAVAILABLE
+    - DEADLINE_EXCEEDED
+  - name: non_idempotent
+    retry_codes: []
+  retry_params_def:
+  - name: default
+    initial_retry_delay_millis: 100
+    retry_delay_multiplier: 1.3
+    max_retry_delay_millis: 60000
+    initial_rpc_timeout_millis: 60000
+    rpc_timeout_multiplier: 1
+    max_rpc_timeout_millis: 60000
+    total_timeout_millis: 600000
+  methods:
+  - name: SetBreakpoint
+    flattening:
+      groups:
+      - parameters:
+        - debuggee_id
+        - breakpoint
+        - client_version
+    required_fields:
+    - debuggee_id
+    - breakpoint
+    - client_version
+    request_object_method: true
+    retry_codes_name: non_idempotent
+    retry_params_name: default
+    timeout_millis: 60000
+  - name: GetBreakpoint
+    flattening:
+      groups:
+      - parameters:
+        - debuggee_id
+        - breakpoint_id
+        - client_version
+    required_fields:
+    - debuggee_id
+    - breakpoint_id
+    - client_version
+    request_object_method: true
+    retry_codes_name: idempotent
+    retry_params_name: default
+    timeout_millis: 60000
+  - name: DeleteBreakpoint
+    flattening:
+      groups:
+      - parameters:
+        - debuggee_id
+        - breakpoint_id
+        - client_version
+    required_fields:
+    - debuggee_id
+    - breakpoint_id
+    - client_version
+    request_object_method: true
+    retry_codes_name: idempotent
+    retry_params_name: default
+    timeout_millis: 60000
+  - name: ListBreakpoints
+    required_fields:
+    - debuggee_id
+    - include_all_users
```

Comment:
> Is that actually what the generated library enforces? E.g. the user can't pass 0 for required numbers, "" for required strings, and [] for required bytes?

No, it doesn't enforce the restriction. It just makes those parameters mandatory in languages (like Python) that distinguish required from keyword/optional arguments, making it less likely for users to leave them unset accidentally. AFAIK, there's no technical reason that we couldn't validate, though.

> From inspecting the generated config - the tool is just assuming all fields of the requests are required?

Yes. As I said before, a better heuristic would be helpful.

> FWIW, an HTTP pattern that uses GET tells you with full confidence that the fields that don't appear in the pattern are optional (as HTTP GET requests can't have bodies). And pretty much full confidence that string fields that appear in the pattern are required (as in "can't be empty").

Right, unless someone is silly and decides to make a query parameter semantically required. If you're interested in improving how we handle required/optional params, please coordinate with @saicheems, since I think he's already started working on that.